core/tx_pool.go

```diff
@@ -1227,6 +1252,15 @@ func (pool *TxPool) runReorg(done chan struct{}, reset *txpoolResetRequest, dirt
 	// remove any transaction that has been included in the block or was invalidated
 	// because of another transaction (e.g. higher gas price).
 	if reset != nil {
+		if reset.newHead != nil {
+			var txHashes []common.Hash
+			addBlock := pool.chain.GetBlock(reset.newHead.Hash(), reset.newHead.Number.Uint64())
+			for _, transaction := range addBlock.Transactions() {
+				txHashes = append(txHashes, transaction.Hash())
+			}
+			pool.calculateTxsLifecycle(txHashes, addBlock.Time())
```

Comment:
with relaxed block times, we might have block times shorter than a second. So, we will need a finer resolution then a second.

Reply:
We can just use `time.Now()`instead of block timestamp here.

Reply:
make sense